src/txn/txn_rollback_to_stable.c

```diff
@@ -125,6 +125,9 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *first
 
             upd->txnid = WT_TXN_ABORTED;
             WT_STAT_CONN_INCR(session, txn_rts_upd_aborted);
+
+            if (F_ISSET(upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS))
+                oldest_unstable_hs_upd = upd;
```

Comment:
Should we here just use a flag instead of storing the update?

Reply:
We cannot use this `rollback_timestamp `? Why?

Reply:
How will removing things this way in hs impacting reconstructing reverse modifies?

Reply:
In the following scenario, stable timestamp (30), if we remove all the entries with a stable timestamp.
update list - U7 (70) -> U6 (60) -> U5 (50)
on-disk page - U4 (40)
History store - U6(60), U5 (50), U4 (40), U3(30), U2 (20), U1(10)

We remove all the entries till U2 in the history store by passing the stable timestamp. And later while processing the on-disk entry U4(40) we find U2 and retrieve it instead of U3.

Reply:
ok, the fix is tricky.

Reply:
At least we need a comment for this.

Reply:
ok.

Reply:
Would passing `rollback_timestamp + 1` work here? It looks like the issue is that `__rollback_delete_hs` deletes all entries up to and including the the provided timestamp.
I like the idea of removing everything newer than `stable_ts` here. It means we don't need to worry about missing any edge cases from trimming the update chain.

Reply:
No, it doesn't work. It works if the on-disk version of the key is stable. In case it is not stable, to replace the on-disk version, we need to traverse the history store again. If we remove all the updates in the first pass during the update chain, we don't find the proper stable update in the history store in the second pass and it can lead to the problem.

First pass - calling delete_hs while processing the update chain.
Second pass - Fixing the unstable on-disk entry.

Reply:
This makes me feel we should fix this in the insert list code path instead of here.

Reply:
The main problem with this approach is we don't have the information of whether the on-disk key is stable or not. Moving this logic to the `insert list` can fix for the row-store. But it doesn't work for the column store.

Reply:
Yes, we need some special logic for column store but that should be ok I guess. We only need to pass to the insert list function whether we have a on-disk value or not.

Reply:
It sounds like we'd need to do some further investigation before we move this change to the insert list handling.

Given this is an interim fix until WT-10017 is ready I reckon we're ok to deliver as is. @quchenhao I've update the comment to mention why we don't use `stable_ts`/`rollback_timestamp`.